### ISSUE TITLE
Wrong SQL generated for PostgreSQL functions returning single UDT OUT parameters

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12659]
It seems we currently don't support calling a stored function that returns a UDT `OUT` parameter in PostgreSQL. E.g.

```sql
CREATE FUNCTION p_enhance_address2 (address OUT u_address_type)
AS $$
BEGIN
	address := (
		SELECT t_author.address
		FROM t_author
		WHERE first_name = 'George'
	);
END;
$$ LANGUAGE plpgsql;
```

Calling this routine generates the following query:

```sql
select "address"
from "public"."p_enhance_address2"()
```

But there is no `address` column in the result, if there's only a single UDT `OUT` parameter. With several `OUT` parameters, it would work as expected. So, as a workaround, we could run this query instead:

```sql
select "t"::u_address_type as "address"
from "public"."p_enhance_address2"() as "t"
```

**Comment by lukaseder** [comment]
We have an integration test that fails because it returns a table row type containing a *column* t, which conflicts with the generated *table* t...

**Comment by lukaseder** [comment]
So, this works better:

```sql
select row("t".*) as "address"
from "public"."p_enhance_address2"() as "t"
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.5 (https://github.com/jOOQ/jOOQ/issues/12661)

**Comment by lukaseder** [comment]
It appears this workaround can be removed again in jOOQ 3.22:

- https://github.com/jOOQ/jOOQ/issues/20082
